### PR TITLE
resource-manager: supports redirect the http request to primary server

### PR DESCRIPTION
**Pull request by nolouch**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/6051

### What is changed and how does it work?

```commit-message
resource-manager: supports redirect the HTTP request to primary server
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

before:
image

after:
image

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by nolouch** [comment]
ptal @disksing @CabinfeverB @rleungx

File: pkg/utils/apiutil/apiutil.go [hunk 1]
```diff
@@ -367,3 +378,75 @@ func RegisterUserDefinedHandlers(registerMap map[string]http.Handler, group *API
...
+			log.Error("request failed", errs.ZapError(errs.ErrSendRequest, err))
+			continue
+		}
+		defer resp.Body.Close()
+		var reader io.ReadCloser
+		switch resp.Header.Get("Content-Encoding") {
+		case "gzip":
+			reader, err = gzip.NewReader(resp.Body)
+			if err != nil {
+				log.Error("failed to parser response with gzip compress", zap.Error(err))
```
Comment (rleungx) [inline pkg/utils/apiutil/apiutil.go:415]:
```suggestion
				log.Error("failed to parse response with gzip compress", zap.Error(err))
```

**Review by rleungx — APPROVED** [review]
The rest LGTM

File: pkg/utils/apiutil/apiutil.go [hunk 2]
```diff
@@ -367,3 +378,75 @@ func RegisterUserDefinedHandlers(registerMap map[string]http.Handler, group *API
...
+				continue
+			}
+			defer reader.Close()
```
Comment (HuSharp) [inline pkg/utils/apiutil/apiutil.go:418]: if reader meets error, won't it close the reader?

**Comment by nolouch** [comment]
ptal @CabinfeverB  @HuSharp

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**